test/shared/src/main/scala/zio/test/mock/MockSpec.scala

```diff
@@ -95,7 +95,7 @@ sealed trait MockSpec[-R, +E, +A] { self =>
 
         case MockSpec.FlatMap(nestedSpec, continue) =>
           for {
-            _   <- state.continuationsRef.update(continue :: _)
+            _   <- state.continuationsRef.update(continue.asInstanceOf[Any => MockSpec[R, E, Any]] :: _)
```

Comment:
Add a comment about Dotty?

Reply:
I think Dotty is actually right here. The type of `continue` is `A => MockSpec[R, E, Any]` for some concrete type `A`. So it isn’t sound to infer this as `Any => MockSpec[R, E, Any]`. It is only safe for us to do so because we know that we have built the list of continuations in such a way that the types match up.

Reply:
Right, upcasting on the input is actually downcasting.